Courseography/courseography
# Refactored Sidebar from Haskell to a React Component

## Motivation and Context
This PR removes the previous `Text.Blaze.Html5 Haskell` code that was written to render the sidebar HTML for Courseography and replaces it with React components.
To do this, @stefurac and I created a `Focus.js` and `Sidebar.js` to replace the Haskell code, as well as added a `Container.js` component which is a parent component of the `Sidebar.js` and `Graph.js` components.


## Your Changes
- Created `Container.js` component
- Created `Sidebar.js` component
- Created `Focus.js` component
- Reduced the code in `main.js`
- Removed `sidebar-divs.js` as the previously existing jQuery methods are now converted to React.
- Removed previously written Haskell code in Graph.hs to render the sidebar

**Type of change** (select all that apply):
- [x] Refactoring (internal change to codebase, without changing functionality)
- [ ] Test update (change that modifies or updates tests only)


## Testing
#### Setup
- Run `stack build`
- Run `stack exec courseography css` to load new css
- Run `stack exec courseography`
- Navigate to `http://localhost:8000/graph`
#### Manual Testing
Do the following steps for manual testing. Keep the console open the entire time to ensure there are no errors.
1. Click the sidebar and ensure it opens correctly. All the graph buttons in the sidebar should load properly. Also ensure the arrow favicon flips.
2. Click a non-computer science graph. The graph should load properly and the focus button in the sidebar should not be clickable.
3. Refresh the page. The graph you were previously on should show up and the focuses tab should still be disabled.
4. Click the computer science graph. Click on a node. The sidebar should automatically close. Click on a couple more nodes. Then open the sidebar and click reset graph. This should reset the nodes and the FCE count as expected.
5. Click on Focuses. Click on a particular focus. The focuses should be highlighted in the graph and the focus details should expand and be readable. Navigate to different focuses and ensure the focuses in the graph change. Click the same focus to undo the focus effect.
6. Click the export button on the top. This should open and there should be a GET request for an image in the console. However this does not work locally for some reason (tested on master and same error occurs). Ensuring the GET request for the image is sent should be sufficient.


## Questions and Comments (if applicable)
 - We should make sure that the export modal functionality works before deploying.
 - In order for the CSS changes to propagate, you may need to clear browser cache.

## Proposed Future Updates
 - When a focus is selected, we could disable the graphs navigation. Otherwise, the user is prone to clicking on another graph and the graph is unresponsive because the previous focus is still "focused".
 - When a focus is selected, clicking reset graph should unfocus.
 - The graph and focus buttons go off the sidebar "page"/div and into the disclaimer. This happens in the deployed courseography as well. This is especially obvious when I was working with the console up, but it isn't a huge deal.
 - Add animation to the sidebar opening to match previous animation.

## Checklist

- [ ] I have verified that the CircleCI tests have passed. 
- [ ] I have updated the Changelog.md file. 


### Required documentation changes (if applicable)
????

## TODO:
 - [x] Update reset button to reset focuses
 - [x] Add tests for sidebar
